docs/src/pages/concepts/styling.mdx

```diff
@@ -0,0 +1,401 @@
+
+```tsx
+export const Component = () => (
+  <Box
+    css={{
+      width: ['100%', null, 'small'],
+    }}
+  >
+    I am a reposnsive Box!
+  </Box>
+);
+```
+
+## Styling Components
+
+As mentioned before, Marigold uses a specific format to define the appearancesce of components. It consists of the `base`, `variant` and `size` (see [Component Styles](#component-styles)). The most basic API for styling a component looks like this:
+
+```ts
+// Single part styles for a <Button>
+export const Button = {
+  // Styles for the default appearance
+  base: {},
+  // Styles for the visual style variations
+  variant: {},
+  // Styles for the size variations
+  size: {},
+};
+```
+
+Following this format you can style Marigold's component (with the exception of [layout components](/concepts/layouts/)). Styles will be scoped to the component. They will not leak into any other component.
+
+As you can see in the following example styling the `<Button>` component, the `base` styles are always applied. `variant` and `size` can be added via the corresponding props of the `<Button>`. Adding these two props is optional, since every component has at least one base appearance.
+
+```tsx preview file=concepts/simple-styling.demo.tsx
+
+```
+
+There are components which are more complex than a button, a table for example. They consist of more than one part. Every table has cells, rows, a body for example. With some additional parts like a header.
+
+Marigold's styling solution also allows to handle styling of complex component, which internally have more than one part. Similar to a single part component, the format uses `base`, `variant` and `size`. The parts of a component (`header`, `row`, `cell`) are the child props of `base`, `variant` and `size`. Styling is then applied like with a single component. The parts will always have their base style and in addition, can have a `variant` and/or a `size`.
+
+The `variant` and `size` props are usually added on the top most component (`<Table>`) and will then be passed down to their children. For example, if you want collapsed rows add the variant to the table, `<Table variant="collapse">`. These composite components usually share variants and sizes between parts so that the different parts work well together.
+
+```ts
+// Single multi part styles for a <Table>
+export const Table = {
+  // Styles for the default appearance
+  base: {
+    // <Table.Header>
+    header: {},
+    // <Table.Row>
+    row: {},
+    // <Table.Cell>
+    cell: {},
+  },
+  // Styles for the visual style variations
+  variant: {
+    head: {},
+    row: {},
+    cell: {
+      // <Table.Cell variant="collapsed">
+      collapsed: {},
+    },
+  },
+  // Styles for the size variations
+  size: {
+    head: {
+      // <Table.Header size="large">
+      large: {},
+    },
+    row: {},
+    cell: {},
+  },
+};
+```
+
+### Using the `useComponentStyles` hook
```

Comment:
useComponentStyles don't appear in the Toc, you have to find another name or remove the `